src/vm/exceptionhandling.cpp

```diff
@@ -4336,17 +4400,20 @@ VOID DECLSPEC_NORETURN UnwindManagedException(EXCEPTION_RECORD* exceptionRecord,
         }
         else
         {
-            controlPc = Thread::VirtualUnwindLeafCallFrame(currentFrameContext);
+            Thread::VirtualUnwindLeafCallFrame(currentFrameContext);
+        }
+
+        // Check whether we are crossing managed-to-native boundary
+        if (!ExecutionManager::IsManagedCode(currentFrameContext->Rip))
+        {
+            // Return back to the UnwindManagedExceptionPass1 and let it unwind the native frames
+            return;
         }
 
-        //
-        //TODO: check whether we are crossing managed-to-native boundary
-        // and add support for this case.
-        //
     } while (IsSpInStackLimits(currentFrameContext->Rsp, stackLowAddress, stackHighAddress) &&
         (establisherFrame != targetFrameSp));
 
-    printf("UnwindManagedException: Unwinding failed. Reached the end of the stack.\n");
+    printf("UnwindManagedExceptionPass2: Unwinding failed. Reached the end of the stack.\n");
```

Comment:
Makes sense. I've also changed it the same way in the UnwindManagedExceptionPass1